src/components/QueryEditor/QueryEditorForm.tsx

```diff
@@ -195,3 +210,37 @@ export function QueryEditorForm({ query, onChange, datasource, onRunQuery, group
     </div>
   );
 }
+
+const getLinksStyles = stylesFactory(() => ({
+  container: css`
+    display: flex;
+  `,
+  link: css`
+    white-space: nowrap;
+  `,
+}));
+type XrayLinksProps = {
+  datasource: XrayDataSource;
+  query: XrayQuery;
+  range?: TimeRange;
+};
+function XrayLinks({ datasource, query, range }: XrayLinksProps) {
```

Comment:
I know it is a very last minute fix but you already refactored this so nicely. Can this component live in its own file?